python/src/worker/tasks/factory.py

```diff
@@ -78,6 +96,4 @@ def _factory(self, msg) -> Task:
         try:
             return self.tasks[task_name](msg)
         except KeyError as e:
-            raise ValueError(
-                f"Task class with name {task_name} was not found"
-            ) from e
+            raise ValueError(f"Task class with name {task_name} was not found") from e
```

Comment:
changing the exception type in the handling looks weird, it still is a `KeyError` not a `ValueError`. I would either
`raise KeyError(f"Task class with name {task_name} was not found") from e`
or
`raise KeyError(f"Task class with name {task_name} was not found: {e}")`

Reply:
Changed